### DISCUSSION TITLE
Embedded timestamps to Point Grey PGE Blackfly 50S5C cameras

### DISCUSSION THREAD

**Discussion by MariaBjoernestad** [bonsai-rx/bonsai#1825] [Q&A]
Hi,
We have images from two Point Grey PGE Blackfly 50S5C cameras, and I'm trying to take out the embedded timestamps from the first four pixels in the images.  I'm uising the example code found at https://www.flir.eu/support-center/iis/machine-vision/knowledge-base/imaging-products-timestamping-and-different-timestamp-mechanisms/, which look like (in cpp):

imageTimeStampToSeconds(unsigned int uiRawTimestamp)
{

   int nSecond      = (uiRawTimestamp >> 25) & 0x7F;   // get rid of cycle_* - keep 7 bits
   int nCycleCount  = (uiRawTimestamp >> 12) & 0x1FFF; // get rid of offset    13
   int nCycleOffset = (uiRawTimestamp >>  0) & 0xFFF;  // get rid of *_count   12

   return (double)nSecond + (((double)nCycleCount+((double)nCycleOffset/3072.0))/8000.0);
}

The calculated timestamps are not correct, that is, the timestamps are not sorted in the right way. Any suggestions to do this in a better way?

**Comment by bruno-f-cruz** [comment]
I assume you are saving the pictures with Bonsai? Have you tried to use the spinnaker software to validate the approach first? I am not sure if, depending on the encoding you use, those pixel's information will be preserved.

Alternatively, you can use the metadata attached to each frame that is exposed via the `SpinnakerCapture` operator.

**Reply by MariaBjoernestad** [reply]
Yes, I believe so. Thanks for the feedback, I'll try that out.
